docs/assets/js/nodes.js

```diff
@@ -0,0 +1,144 @@
+    const nodeId = item.getAttribute("id");
+    if (selectedNode === nodeId) {
+      item.classList.add("active");
+    } else {
+      item.classList.remove("active");
+    }
+  });
+};
+
+const toggleVisibleAttributes = () => {
+  q(".secondary-container").classList.remove("active");
+  if (!selectedNode) return;
+  q(".secondary-container").classList.add("active");
+
+  const documentNode = q(`.accordion-item[node_id="${selectedNode}"]`);
+  const attributeIds = documentNode.getAttribute("attribute_ids");
+  const attributeList = attributeIds.split(",");
+
+  qq(".attribute-container").forEach((attribute) => {
+    const attributeId = attribute.getAttribute("id");
+    if (attributeList.includes(attributeId)) {
+      attribute.classList.add("active");
+    } else {
+      attribute.classList.remove("active");
+    }
+  });
+};
+
+const toggleAttributeSelected = (event) => {
+  const attributeElement = event.currentTarget.parentNode;
+  attributeElement.classList.toggle("selected");
+};
+
+const setNodeQueryParam = (nodeId) => {
+  const url = new URL(window.location);
+  if (nodeId != null) {
+    url.searchParams.set(nodeQueryParamKey, nodeId);
+  } else {
+    url.searchParams.delete(nodeQueryParamKey);
+  }
+  window.history.pushState({}, "", url);
+};
+
+const renderPage = () => {
+  toggleExpandedCategories();
+  toggleSelectedCategory();
+  toggleVisibleAttributes();
+  toggleVisibleCategory();
+};
+
+const toggleNode = (nodeId, depth) => {
+  if (selectedNodes[depth] === nodeId) {
+    delete selectedNodes[depth];
+    selectedNode = selectedNodes[depth - 1];
+  } else {
+    selectedNodes[depth] = nodeId;
+    selectedNode = nodeId;
+  }
+  Object.keys(selectedNodes).forEach((key) => {
+    if (key > depth) {
+      delete selectedNodes[key];
+    }
+  });
+
+  setNodeQueryParam(selectedNode);
+  renderPage();
+};
+
+const setupListeners = () => {
+  qq(".accordion-item").forEach((item) => {
+    item.addEventListener("click", (e) => {
+      toggleNode(
+        e.target.getAttribute("node_id"),
+        e.target.closest(".sibling-list").getAttribute("node_depth")
+      );
+    });
+  });
+  qq(".attribute-title").forEach((attribute) => {
+    attribute.addEventListener("click", toggleAttributeSelected);
+  });
+};
+
+const setInitialNode = () => {
+  const initialNode = getQueryParam(nodeQueryParamKey);
+  if (!initialNode) return;
+
+  const documentNode = q(`.accordion-item[node_id="${initialNode}"]`);
+  if (!documentNode) return;
+
+  const ancestors = documentNode.getAttribute("ancestor_ids")
+    ? documentNode.getAttribute("ancestor_ids").split(",")
+    : [];
+  const depth = ancestors.length;
+
+  ancestors.forEach((ancestor, index) => {
+    selectedNodes[depth - index - 1] = ancestor;
+  });
+  selectedNodes[depth] = initialNode;
+  selectedNode = initialNode;
+};
+
+export const resetToCategory = (categoryId) => {
+  selectedNodes = {};
+  setNodeQueryParam(categoryId);
+  setInitialNode();
+  renderPage();
+};
```

Comment:
this module is a straight copy of what was in index.js before related to nodes, except for this one function that I added which is the little bit of logic that search.js needs.